### PR TITLE
Pagespeed configuration.

### PR DESCRIPTION
**Pull request by efpapado**
Based on the instructions https://www.modpagespeed.com/doc/downstream-caching

### CODE REVIEW

File: templates/defaults/vcl_recv.vcl.tmpl [hunk 1]
```diff
@@ -57,4 +57,23 @@ sub vcl_recv {
...
+    {{ if getenv "VARNISH_PAGESPEED_SECRET_KEY" }}
+    # Tell PageSpeed not to use optimizations specific to this request.
+    set req.http.PS-CapabilityList = "fully general optimizations only";
+
+    # Don't allow external entities to force beaconing.
+    unset req.http.PS-ShouldBeacon;
+
+    # Authenticate the purge request by IP.
+    if (req.method == "PURGE") {
+        if (!client.ip ~ purge) {
```
[the remark this answers is not in the document]
Comment (efpapado) [reply templates/defaults/vcl_recv.vcl.tmpl]: What should be the IP range? I found some examples for Docker setups:

```
acl purge {
  "localhost";
  "127.0.0.1";
  "172.17.0.0/16";
}
```

Also, do you think `templates/includes/purge.vcl.tmpl` is a proper path to add?

[the remark this answers is not in the document]
Comment (csandanov) [reply templates/defaults/vcl_recv.vcl.tmpl]: Right, I guess we don't need this purge logic here at all, we already have all authentication implemented in `purge.vcl`

[the remark this answers is not in the document]
Comment (efpapado) [reply templates/defaults/vcl_recv.vcl.tmpl]: OK, I removed it.

File: templates/defaults/vcl_recv.vcl.tmpl [hunk 2]
```diff
@@ -57,4 +57,19 @@ sub vcl_recv {
...
+    ### Pagespeed
+    # Based on the suggestions https://www.modpagespeed.com/doc/downstream-caching
+    {{ if getenv "VARNISH_PAGESPEED_SECRET_KEY" }}
+    # Tell PageSpeed not to use optimizations specific to this request.
+    set req.http.PS-CapabilityList = "fully general optimizations only";
+
+    # Don't allow external entities to force beaconing.
+    unset req.http.PS-ShouldBeacon;
+
+    if (req.method == "PURGE") {
```
Comment (csandanov) [inline templates/defaults/vcl_recv.vcl.tmpl]: We don't need these purge related lines at all here

Comment (efpapado) [reply templates/defaults/vcl_recv.vcl.tmpl]: OK removed.

**Pull request merged**